src/widgets/dialogs/LogsPopup.cpp

```diff
@@ -0,0 +1,151 @@
+#include "LogsPopup.hpp"
+
+#include "IrcMessage"
+#include "common/NetworkRequest.hpp"
+#include "providers/twitch/TwitchChannel.hpp"
+#include "providers/twitch/TwitchMessageBuilder.hpp"
+#include "widgets/helper/ChannelView.hpp"
+
+#include <QDateTime>
+#include <QMessageBox>
+#include <QVBoxLayout>
+
+namespace chatterino {
+
+LogsPopup::LogsPopup()
+{
+    this->initLayout();
+    this->resize(400, 600);
+}
+
+void LogsPopup::initLayout()
+{
+    {
+        QVBoxLayout *layout = new QVBoxLayout(this);
+        layout->setMargin(0);
+
+        {
+            this->channelView = new ChannelView(this);
+            layout->addWidget(this->channelView);
+        }
+
+        this->setLayout(layout);
+    }
+}
+
+void LogsPopup::setInfo(ChannelPtr channel, QString userName)
+{
+    this->channel_ = channel;
+    this->userName = userName;
+    this->setWindowTitle("Logs for " + this->userName + " in " + this->channel_->name +
+                         "'s history");
```

Comment:
`User's logs in #channel` would be more compact and brings the most important info to the start of the message.